### ISSUE TITLE
OOM issue when using `final_activation` in `MLP`

### ISSUE THREAD

**Issue by pablo2909** [patrick-kidger/equinox#253]
Hello,

I am using an MLP as part of a continuous normalizing flow implementation. As the base case the `MLP` does not use a `final_activation` and it runs smoothly. However if I apply `jnn.sigmoid`as a final activation I face an OOM issue. I am not sure where this comes from but I suspect it has to do with `jax.vjp`. Also, I am not sure how to provide a minimal working example. Thus, does someone have an idea on the origin of issue and/or how I can explore it more?

Thank you :)

Note: I am not exactly sure if this belongs here, let me know if it is better suited to open an issue on the jax repo. :)

**Comment by patrick-kidger** [comment]
How close to an OOM are you without the `final_activation`? The small amount of extra memory needed to handle the `final_activation` may simply be "the straw that broke the camel's back".

**Comment by pablo2909** [comment]
Hi @patrick-kidger 

Sorry for my delayed response, I was going through important changes in the code and unfortunately can't reproduce it...

The only comment I can make now is on the running time. Everything else kept constant, an epoch with final activation takes twice as long, from ~100s to ~200s. 

I understand it is hard to make any productive comment right now, I will try to come up with a reproducible code in the coming days, otherwise I'll just close the issue :) 

Thank you
